app/controllers/users_controller.rb

```diff
@@ -27,23 +27,20 @@ def index
 
   def show
     @abilities = Ability.on_user(@user)
-    # allposts definition copied from posts() method. posts is limited to 15.
-    @allposts = if current_user&.privilege?('flag_curate') || @user == current_user
-                  Post.all
+
+    all_posts = if current_user&.privilege?('flag_curate') || @user == current_user
+                  @user.posts
                 else
-                  Post.undeleted
-                end.where(user: @user).list_includes.joins(:category)
+                  @user.posts.undeleted
+                end
+                .list_includes
+                .joins(:category)
                 .where('IFNULL(categories.min_view_trust_level, 0) <= ?', current_user&.trust_level || 0)
                 .user_sort({ term: params[:sort], default: :score },
                            age: :created_at, score: :score)
-                .paginate(page: params[:page], per_page: 25)
-    @posts = if current_user&.privilege?('flag_curate')
-               @user.posts
-             else
-               @user.posts.undeleted
-             end.list_includes.joins(:category)
-             .where('IFNULL(categories.min_view_trust_level, 0) <= ?', current_user&.trust_level || 0)
-             .order(score: :desc).first(15)
+
+    @posts = all_posts.first(15)
```

Comment:
Much nicer, thanks!

Reply:
Since I missed the problem in my review of the previous pull request, and we've both been working on this code in different places, it might be worth getting a third pair of eyes to review this pull request.